TLM/TLM/Manager/Impl/TrafficLightSimulationManager.cs

```diff
@@ -219,15 +219,17 @@ public class TrafficLightSimulationManager
                                    TrafficLightState pedestrianLightState,
                                    bool vehicles,
                                    bool pedestrians) {
-            // stock code from RoadBaseAI.SetTrafficLightState
+            // STOCK-CODE START from RoadBaseAI.SetTrafficLightState
             int num = (int)pedestrianLightState << 2 | (int)vehicleLightState;
             if (segmentData.m_startNode == nodeId) {
                 if ((frame >> 8 & 1u) == 0u) {
+                    // update 'x' bits (int)[0000 0000 0000 xxxx]
+                    // -16 / FFF0 / [1111 1111 1111 0000]
```

Comment:
what is `0xFFF0`? maybe you can define local variable `const byte variablename = 0xFFF0` for more readability.

Reply:
same thing for all the rest of the constants.

Reply:
it's decoded representation what is going on and how game stores traffic light state. Nothing more than that. We lived without that info quite a long time so it's a bonus. Code is copy-pasted, 100% vanilla but we cannot call it because we are patching it to be skipped 😄

Reply:
but its probably out of scope of the PR

Reply:
I've had to apply changes from vanilla (they extended value from `ushort` to `uint`) and magic bit shift values also changed, so I thought it would be nice to add explanation what is going on there 😉 That's all